src/draft/draftRuntime.ts

```diff
@@ -0,0 +1,114 @@
+import * as vscode from 'vscode';
+import { LoggingDebugSession, InitializedEvent, TerminatedEvent } from 'vscode-debugadapter';
+import { EventEmitter } from 'events';
+import { ChildProcess, exec, execFile, spawn } from 'child_process';
+import { OutputChannel } from 'vscode';
+import { Readable } from 'stream';
+import { host } from '../host';
+import { shell } from '../shell';
+import { fs } from '../fs';
+import { create as draftCreate } from './draft';
+
+export class DraftRuntime extends EventEmitter {
+	private _draft = draftCreate(host, fs, shell);
+	private _connectProccess: ChildProcess;
+
+	constructor() {
+		super();
+	}
+
+	public killConnect() {
+		this._connectProccess.kill('SIGTERM');
+	}
+
+	public async draftUpDebug(config: vscode.DebugConfiguration) {
+
+		let output = vscode.window.createOutputChannel("draft");
+		output.show();
+
+		let isPresent = await this._draft.checkPresent();
+		if (!isPresent) {
+			host.showInformationMessage("Draft is not installed!");
+			return;
+		}
+
+		if (!this._draft.isFolderMapped(vscode.workspace.rootPath)) {
+			host.showErrorMessage("This folder does not contain a Draft app. Run draft create first!");
+			return;
+		}
+
+		// //wait for `draft up` to finish
+		await waitForProcessToExit(createProcess('draft', ['up'], output));
+
+		// wait for `draft connect` to be ready
+		this._connectProccess = createProcess('draft', ['connect'], output);
+		await waitConnectionReady(this._connectProccess);
+
+		host.showInformationMessage(`attaching debugger`);
+
+		let cfg: vscode.DebugConfiguration = config['original-debug'];
+
+		// issue setting localRoot to worspaceFolder in nested config
+		//
+		// see https://github.com/Microsoft/vscode-node-debug/issues/178
+		cfg['localRoot'] = vscode.workspace.rootPath;
```

Comment:
This is true. However, the reason I'm currently hardcoding that value is because you would usually use `${workspaceFolder}` - however, that is not available in nested configurations.

The workaround for this seems to: have the active workspace as the default value and check if the user overwrote it - but you have to use the absolute path it seems.